llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -2510,6 +2510,7 @@ void SpirvLowerGlobal::lowerBufferBlock() {
         DominatorTree &domTree = m_functionAnalysisManager->getResult<DominatorTreeAnalysis>(*func);
 
         SmallVector<Instruction *, 8> usesToReplace;
+        SmallVector<Instruction *, 8> unreachableUses;
```

Comment:
Actually, despite what I said earlier, you might be able to do this without the unreachableUses vector as it is already saving the ones it wants to process into usesToReplace. Not a blocker though.

Reply:
I think you do need to do it this way - it's still replacing a use ultimately (I tried and it didn't work).

Reply:
However, tests show up a case that isn't handled (all uses replaced with poison and insertPoint is still nullptr). So a slight re-arrangement and a null test are required I think.

Reply:
Is it the case that, having processed all the reachable uses in instructions, all that is left are unreachable uses in instructions? I.e. have you also got rid of non-instruction uses by then?

If so, then you can just pick one off the front each time in a loop:
```
while (!global.use_empty())
  *global.use_begin() = PoisonValue::get(global.use_begin()->getType());
```
(pointers and refs are approximate only as I can never remember how to use a Use properly).

Reply:
Hmm, could be. I think the implementation I've ended up with is clear - unless you think it needs refactoring to be more like what you suggest, I'll leave as-is?